src/test/java/com/amazon/ion/impl/bin/IonRawBinaryWriterTest_1_1.kt

```diff
@@ -0,0 +1,230 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package com.amazon.ion.impl.bin
+
+import com.amazon.ion.IonException
+import com.amazon.ion.IonType
+import java.io.ByteArrayOutputStream
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.CsvSource
+
+class IonRawBinaryWriterTest_1_1 {
+
+    private inline fun writeAsHexString(block: IonRawBinaryWriter_1_1.() -> Unit): String {
+        val baos = ByteArrayOutputStream()
+        val rawWriter = IonRawBinaryWriter_1_1(
+            out = baos,
+            buffer = WriteBuffer(BlockAllocatorProviders.basicProvider().vendAllocator(32)),
+            lengthPrefixPreallocation = 1,
+        )
+        block.invoke(rawWriter)
+        @OptIn(ExperimentalStdlibApi::class)
+        return baos.toByteArray().joinToString(" ") { it.toHexString(HexFormat.UpperCase) }
+    }
+
+    private inline fun assertWriterOutputEquals(hexBytes: String, block: IonRawBinaryWriter_1_1.() -> Unit) {
+        assertEquals(hexBytes, writeAsHexString(block))
+    }
+
+    @Test
+    fun `calling close while in a container should throw IonException`() {
+        assertThrows<IonException> {
+            writeAsHexString {
+                stepInList(false)
+                close()
+            }
+        }
+    }
+
+    @Test
+    fun `calling finish while in a container should throw IonException`() {
+        assertThrows<IonException> {
+            writeAsHexString {
+                stepInList(true)
+                finish()
+            }
+        }
+    }
+
+    @Test
+    fun `calling stepOut while not in a container should throw IonException`() {
+        assertThrows<IonException> {
+            writeAsHexString {
+                stepOut()
+            }
+        }
+    }
+
+    @Test
+    fun `calling writeIVM when in a container should throw IonException`() {
+        assertThrows<IonException> {
+            writeAsHexString {
+                stepInList(false)
+                writeIVM()
+            }
+        }
+    }
+
+    @Test
+    fun `calling finish should cause the buffered data to be written to the output stream`() {
+        val actual = writeAsHexString {
+            writeIVM()
+            finish()
+        }
+        // Just checking that data is written, not asserting the content.
+        assertTrue(actual.isNotBlank())
```

Comment:
Is that the behavior we want for Ion 1.1, though? What if someone calls `writeIVM()` and then `finish()` (maybe they have logic that writes after every top level piece of data) and then try to write Ion 1.1 data? I know it's a niche case, but that would cause invalid data to be written.

In addition, because emitting an IVM resets the encoding environment, I think the decision of when to write or not write the IVM belongs in a higher level writer that is aware of the encoding environment.